pandas/core/indexes/api.py

```diff
@@ -220,7 +220,8 @@ def conv(i):
         index = indexes[0]
         for other in indexes[1:]:
             if not index.equals(other):
-                return _unique_indices(indexes)
+                index = _unique_indices(indexes)
+                break
```

Comment:
just a personal preference, but for me something like

```
        if not all(index.equals(other) for other in indexes[1:]):
            index = _unique_indices(indexes)
```
is easier to grok instead of introducing a break.

Reply:
ah, yup that's much more readable. hadn't realized that `all` will short-circuit. will add that in plus some additional tests.